consensus/consensus_service.go

```diff
@@ -101,6 +101,9 @@ func (consensus *Consensus) UpdatePublicKeys(pubKeys []bls_cosi.PublicKeyWrapper
 	consensus.UpdateBitmaps()
 	consensus.ResetState()
 
+	consensus.mutex.Lock()
```

Comment:
This will cause deadlocks in many places where UpdateConsensusInformation() is used.

Reply:
I see.

That was why I introduced smaller lock on view change data struct before.  Now we have one big lock on big function like UpdateConsensusInformation, and the smaller function like ResetViewChangeState can't be locked.  I have to rework a bit on this.